### PR TITLE
Temporarily target test rather than user devnet to dry run parameterized deploys

### PR DESCRIPTION
**Pull request by ognots**
re-uses circleci commands introduced by new nightly deploy process.
uses git tag for docker image and deploy docker tag to devnet.
makes user/test devnet deploy be triggered by git tag

### CODE REVIEW

File: DEPLOY.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+The User release channel is intended as a quasi-stable (no SLA or guarantees at this time) release and testing bed
+for Filecoin Developers, Infrastructure Engineers and Testers.
+
+A release and deploy to the User Devnet is triggered by pushing a git tag meeting the following regex patterns
+- `/^\d+\.\d+\.\d+$/`
+- `/^testnet\-\d+\.\d+\.\d+$/ # @TODO remove this after deploy testing complete`
+
+### Deploy instructions
+
+1. Announce a code freeze
```
Comment (travisperson) [inline DEPLOY.md]: nit: I don't know what a code freeze will do, as we are not branching from master.

File: DEPLOY.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+4. After the `go-filecoin` build successfully completes in CircleCI, you must manually approve the [Terraform deploy workflow](https://circleci.com/gh/filecoin-project/workflows/go-filecoin-infra/tree/filecoin-testnet).
+View the most recent workflow and approve the `hold` job which will subsequently trigger the `deploy_user_devnet`.
+This manual approval gate is to prevent unintentional destructive deployments of the User Devnet.
+@TODO replace the link above with `filecoin-usernet` branch after testing
+
+That's it! The release and deploy process should be complete.
+
+During release testing, bug fixes or changes should be pushed to the release branch and not to master.
+When releasing and deploying bug fixes or changes from the release branch, create and push a new git tag with incremented patch version ie. `0.1.1`
```
Comment (travisperson) [inline DEPLOY.md]: I think the process is to merge changes into master, and then cherrypick them into the release branch, as we need to ensure we don't have regressions.

**Comment by ognots** [comment]
@travisperson what do you mean by the whole release process. More of the humanz part and not just machines?

File: DEPLOY.md (hunk 2 above)
Comment (ognots) [reply DEPLOY.md]: I've reworded this now to reflect this comment

File: DEPLOY.md (hunk 1 above)
Comment (ognots) [reply DEPLOY.md]: remove this completely

File: DEPLOY.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+View the most recent workflow and approve the `hold` job which will subsequently trigger the `deploy_user_devnet`.
+This manual approval gate is to prevent unintentional destructive deployments of the User Devnet.
+@TODO replace the link above with `filecoin-usernet` branch after testing
+
+That's it! The release and deploy process should be complete.
+
+During release testing, bug fixes or changes should be cherry picked from master in to the release branch as needed.
+When releasing and deploying bug fixes or changes from the release branch, create and push a new git tag with incremented patch version ie. `0.1.1`
+
+While release is still in `0.x.x`, it is acceptable for minor version bumps to contain breaking changes. Full semantic versioning will only be enforced on MainNet launch when the version will become `1.0.0.`
```
Comment (ognots) [inline DEPLOY.md]: @travisperson I also just added a bit about versioning assertions. can you sanity check please

**Comment by ognots** [comment]
I'm going to merge this PR so I can test the CircleCI deploy. I have to head out in 35 mins to want to test before then. Will make a separate issue/PR to track DEPLOY.md

**Comment by travisperson** [comment]
@ognots 

The process as described in @phritz document states that the release is deployed to the test devnet first, then is deployed to the user devnet. I was wondering if we wanted to capture this two step process here, or if we wanted to just document the actual deployment to the user devnet.

**Comment by ognots** [comment]
@travisperson okay I think there was some incongruent plans between my doc and phritz's. I didn't have testnet deploy scoped in the related epic or design document, but it will be trivial to add testnet deploys. Will post another PR with changes

**Comment by ognots** [comment]
@travisperson on second thought this bring additional undocumented scope to how testnet release process works. I'd suggest deferring till after launch and keep current flow at nightly and user devnets. We're already introducing a lot of changes

**Pull request merged**